target/linux/ipq806x/base-files/lib/upgrade/openmesh.sh

```diff
@@ -0,0 +1,87 @@
+# The U-Boot loader of the OpenMesh devices requires image sizes and
+# checksums to be provided in the U-Boot environment.
+# The OpenMesh devices come with 2 main partitions - while one is active
+# sysupgrade will flash the other. The boot order is changed to boot the
+# newly flashed partition. If the new partition can't be booted due to
+# upgrade failures the previously used partition is loaded.
+
+platform_do_upgrade_openmesh() {
+	local tar_file="$1"
+	local restore_backup
+	local primary_kernel_mtd
+
+	local setenv_script="/tmp/fw_env_upgrade"
+
+	local kernel_mtd="$(find_mtd_index $PART_NAME)"
+	local kernel_offset="$(cat /sys/class/mtd/mtd${kernel_mtd}/offset)"
+	local total_size="$(cat /sys/class/mtd/mtd${kernel_mtd}/size)"
+
+	# switch to second kernel when flashing to secondary flash region
+	local next_boot_part="1"
+	case "$(board_name)" in
```

Comment:
I have to admit, I'm still not sure if I get why you can't use the bootseq u-boot variable. Furthermore I don't really the hardcoded mtd index.

If I understand the mail you send to me correctly, the u-boot bootseq variable isn't updated if the image set as primary fails to boot. Hence the active image could be the secondary image and if you rely on the bootseq variable you might overwrite the current booted - and only working - secondary image. Am I right?

Reply:
Yes, the problem is that you might overwrite the only working image when the secondary (fallback) image was booted.

Another option would to search for the 0:HLOS index and compare the positions of inactive and 0:HLOS

Reply:
Not really an improvement. Leave it as it is and document somewhere why it has do be done this way (commit message, comment...).